### DISCUSSION TITLE
Merged mining

### DISCUSSION THREAD

**Discussion by gridmaniac** [coinfoundry/miningcore#867] [Ideas]
I would like to know is it possible to add merged mining support for like Dogecoin+Litecoin or Namecoin+Bitcoin without fundamental changes in code?

I read all comments here and @oliverw already wrote about it but I still don't understand,
sorry for double-asking.

**Comment by oliverw** [comment]
Currently not possible but feel free to submit a PR :)

**Comment by dreams-money** [comment]
Hello,

I've been using Miningcore for a few months and love it.  Thank you for putting it together.

I'm writting because I'm interested in implementing merged mining to mining core.  I'm new to C Sharp however!

If I'm not mistaken, the key to merged mining is leveraging the "arbitrary data" field to input in child chain's header. (1)

Looking for any input regarding how to best implement this. 

As far as you know, is this link [(2)](https://github.com/oliverw/miningcore/blob/171a262a3f7861bbb54178d385806b9afebcb0af/src/Miningcore/Blockchain/Bitcoin/BitcoinJob.cs#L223) the "arbitrary data" field they mean? Is this also similar by chance? [(3)](https://github.com/oliverw/miningcore/blob/171a262a3f7861bbb54178d385806b9afebcb0af/src/Miningcore/Blockchain/Bitcoin/BitcoinJob.cs#L578)

I'd love any input if you've looked into this in the past!

(1) https://sec.cs.univie.ac.at/fileadmin/user_upload/i_sec/docs/teaching/thesis/azamyatin_merged_mining.pdf: Page 10 
(2) https://github.com/oliverw/miningcore/blob/171a262a3f7861bbb54178d385806b9afebcb0af/src/Miningcore/Blockchain/Bitcoin/BitcoinJob.cs#L223
(3) https://github.com/oliverw/miningcore/blob/171a262a3f7861bbb54178d385806b9afebcb0af/src/Miningcore/Blockchain/Bitcoin/BitcoinJob.cs#L578
